---
title: eslint-cdk-plugin - no-public-class-fields
titleTemplate: ":title"
---

# no-public-class-fields

<div style="margin-top: 16px; background-color: #595959; padding: 16px; border-radius: 4px;">
  ‚úÖ Using
  <a href="/rules/#recommended-rules">recommended</a>
  in an ESLint configuration enables this rule.
</div>

This rule disallows using class types for public class fields.  
(This rule applies only to classes that extends from `Construct` or `Stack`.)

When class types are used in public fields, it creates tight coupling and exposes mutable state, so not good.

---

#### üîß How to use

```js
// eslint.config.mjs
export default [
  {
    // ... some configs
    rules: {
      "cdk/no-public-class-fields": "error",
    },
  },
];
```

#### ‚úÖ Correct Examples

```ts
import { Construct } from "constructs";
import { IBucket, Bucket } from "aws-cdk-lib/aws-s3";

class MyConstruct extends Construct {
  // ‚úÖ Can use interface public field
  public readonly bucket: IBucket;

  constructor(scope: Construct, id: string) {
    super(scope, id);
    this.bucket = new Bucket(this, "MyBucket");
  }
}
```

#### ‚ùå Incorrect Examples

```ts
import { Construct } from "constructs";
import { Bucket } from "aws-cdk-lib/aws-s3";

class MyConstruct extends Construct {
  // ‚ùå Shouldn't use class type public field
  public readonly bucket: Bucket;

  constructor(scope: Construct, id: string) {
    super(scope, id);
    this.bucket = new Bucket(this, "MyBucket");
  }
}
```
